### PR TITLE
DOC: Improve ``polynomial`` docs

### PR DESCRIPTION
**Pull request by linus-md**
This PR adds several examples and improves spacing at several points.

### CODE REVIEW

File: numpy/polynomial/polynomial.py [hunk 1]
```diff
@@ -508,14 +509,14 @@ def polyder(c, m=1, scl=1, axis=0):
...
     array([  2.,   6.,  12.])
-    >>> P.polyder(c,3) # (d**3/dx**3)(c) = 24
+    >>> P.polyder(c, 3) # (d**3/dx**3)(c) = 24
     array([24.])
-    >>> P.polyder(c,scl=-1) # (d/d(-x))(c) = -2 - 6x - 12x**2
+    >>> P.polyder(c, scl=-1) # (d/d(-x))(c) = -2 - 6x - 12x**2
     array([ -2.,  -6., -12.])
-    >>> P.polyder(c,2,-1) # (d**2/d(-x)**2)(c) = 6 + 24x
+    >>> P.polyder(c, 2, -1) # (d**2/d(-x)**2)(c) = 6 + 24x
     array([  6.,  24.])
```
Comment (linus-md) [inline numpy/polynomial/polynomial.py:520]: I would be in favor of removing these comments. I don't think they provide much value besides the output.

File: numpy/polynomial/polynomial.py [hunk 2]
```diff
@@ -137,9 +137,9 @@ def polyline(off, scl):
     Examples
     --------
     >>> from numpy.polynomial import polynomial as P
-    >>> P.polyline(1,-1)
+    >>> P.polyline(1, -1)
     array([ 1, -1])
-    >>> P.polyval(1, P.polyline(1,-1)) # should be 0
+    >>> P.polyval(1, P.polyline(1, -1)) # should be 0
```
Comment (mdhaber) [inline numpy/polynomial/polynomial.py:142]: While we're fixing PEP8 whitespace,
```suggestion
    >>> P.polyval(1, P.polyline(1, -1))  # should be 0
```
[Inline comments should be separated by at least two spaces from the statement.](https://peps.python.org/pep-0008/#inline-comments)

File: numpy/polynomial/polynomial.py (hunk 1 above)
Comment (mdhaber) [reply numpy/polynomial/polynomial.py:520]: Maybe remove the right hand side; that is pretty obvious from the output. The left hand side may be helpful to confirm the reader's understanding of the arguments.

**Review by mdhaber — APPROVED** [review]
I'll come back to check the new examples more carefully, and if they still look good, I'll go ahead and merge.

File: numpy/polynomial/polynomial.py (hunk 2 above)
Comment (linus-md) [reply numpy/polynomial/polynomial.py]: I changed this at a buch of other locations as well.

**Review by mdhaber — COMMENTED** [review]
*Upon closer review...*
I think that `polyvander2d` and `polyvander3d` are inherently less intuitive than `polyvander`. Still, I think the examples could be more illuminating. @linus-md, now that you understand what the function does, are these particularly good examples for learning to understand what the function does, or are there better ones? Is there any commentary that can be added, or a second/third example to show the effect of input on output, that would help a new user?

**Comment by linus-md** [comment]
@mdhaber what do you say now? I added some calculations to verify the results.

File: numpy/polynomial/polynomial.py [hunk 3]
```diff
@@ -1241,6 +1283,27 @@ def polyvander3d(x, y, z, deg):
...
+                   45.,   1.,   5.,  -3., -15.,   9.,  45.]])
+    
+    To verify some values one can do the following computations:
+    
+    >>> from numpy.polynomial import polynomial as P
+    >>> x, y, z  = np.array([-1, 2, 1]), np.array([1, -2, -3]), np.array([2, 2, 5])
+    >>> deg = [2, 2, 1]
+    >>> V = P.polyvander3d(x=x, y=y, z=z, deg=deg)
+    >>> i, j, k = 0, 1, 1
+    >>> V[:, (deg[1]+1)*(deg[1]+1)*i + (deg[2]+1)*j + k] == x**i * y**j * z**k
```
Comment (mdhaber) [inline numpy/polynomial/polynomial.py:1305]:
```suggestion
    >>> V[:, (deg[1]+1)*(deg[2]+1)*i + (deg[2]+1)*j + k] == x**i * y**j * z**k
```
If I'm reading the documentation correctly. Would be easier to compare the two if `deg` were defined in terms of some variables `l`, `m`, and `n`.

**Review by mdhaber — COMMENTED** [review]
These look more informative. I'll commit my suggestions and merge after checking the rendered documentation.

File: numpy/polynomial/polynomial.py [hunk 4]
```diff
@@ -1241,6 +1283,27 @@ def polyvander3d(x, y, z, deg):
...
+                   -2.,   1.,   2.,   1.,   2.,   1.,   2.],
+                 [  1.,   2.,  -2.,  -4.,   4.,   8.,   2.,   4.,  -4.,  -8.,   8.,
+                   16.,   4.,   8.,  -8., -16.,  16.,  32.],
+                 [  1.,   5.,  -3., -15.,   9.,  45.,   1.,   5.,  -3., -15.,   9.,
+                   45.,   1.,   5.,  -3., -15.,   9.,  45.]])
+    
+    To verify some values one can do the following computations:
+    
+    >>> from numpy.polynomial import polynomial as P
+    >>> x, y, z  = np.array([-1, 2, 1]), np.array([1, -2, -3]), np.array([2, 2, 5])
```
[the remark this answers is not in the document]
Comment (linus-md) [reply numpy/polynomial/polynomial.py:1301]: I will change that above but I need ``np.arrays`` for the exponentiation in line 1305.

[the remark this answers is not in the document]
Comment (mdhaber) [reply numpy/polynomial/polynomial.py]: Hence https://github.com/numpy/numpy/pull/25527#discussion_r1443801109

**Comment by mdhaber** [comment]
A few other things to look out for in the future.

- Lines too long; they cause lint test failure.
- Doctests producing no output but expecting output. When you assign to a variable, `V = ...`, there is no output, so you need to include `>>> V` on the next line if you want to include output in the doctest.
- Doctests expecting output that would result from `str(V)`. Unless you include e.g. `print(V)` or `str(V)` in the doctest, you will get `V.__repr__()` as output, which is different for arrays.

Also, please keep using the appropriate [`[skip ...]` commands](https://numpy.org/doc/stable/dev/development_workflow.html#commands-to-skip-continuous-integration). Looks like they were in the first commit, which is good.

**Pull request merged**